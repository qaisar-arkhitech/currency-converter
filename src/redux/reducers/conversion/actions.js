import {createActionName} from "../../utility"

// conversion name
export const reducerName = "conversion"

// conversion actions type
export const RATES = createActionName(reducerName, "RATES")
