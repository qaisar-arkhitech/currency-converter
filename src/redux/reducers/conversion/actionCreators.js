import {RATES} from "./actions"

// conversion actions
export function createLoadAction() {
  return {
    type: RATES,
    request: {
      type: "get",
      url: "/latest?base=USD",
    },
  }
}
