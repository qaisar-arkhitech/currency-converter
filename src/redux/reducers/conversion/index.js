import {createReducer} from "../../utility"
import {RATES} from "./actions"
import {ERROR, LOADED, LOADING} from "../../middleware/actions"

// conversion initial state
export const initialState = {
  rates: {
    data: null,
    loader: false,
    loadingError: null,
  },
}

// conversion Reducer
const reducers = {
  [RATES + LOADING](state) {
    return {
      ...state,
      rates: {
        ...state.rates,
        data: null,
        loadingError: null,
        loader: true,
      },
    }
  },

  [RATES + LOADED](state, payload) {
    return {
      ...state,
      rates: {
        ...state.rates,
        data: payload,
        loader: false,
      },
    }
  },

  [RATES + ERROR](state, payload) {
    return {
      ...state,
      rates: {
        ...state.rates,
        data: null,
        loader: false,
        loadingError: payload,
      },
    }
  },
}

export default createReducer(reducers, initialState)
