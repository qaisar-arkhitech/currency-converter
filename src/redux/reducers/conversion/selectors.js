import get from "lodash/get"
import {reducerName} from "./actions"

// conversion selectors
export const selectRatesData = (state) =>
  get(state, `${reducerName}.rates.data`)
export const selectRatesLoader = (state) =>
  get(state, `${reducerName}.rates.loader`)
export const selectRatesLoadingError = (state) =>
  get(state, `${reducerName}.rates.loadingError`)
