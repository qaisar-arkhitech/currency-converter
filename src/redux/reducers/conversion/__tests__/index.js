import conversion, {initialState} from ".."

describe("conversion reducer", () => {
  it("should return the initial state", () => {
    expect(conversion(undefined, {})).toEqual(initialState)
    expect(conversion(undefined, {})).toMatchSnapshot()
  })
  it("should handle GET_POST_SUCCESS", () => {})
  it("should handle UPDATE_POST_SUCCESS", () => {})
  it("should handle GET_POST_FAIL", () => {})
  it("should handle GET_POST_START", () => {})
})
