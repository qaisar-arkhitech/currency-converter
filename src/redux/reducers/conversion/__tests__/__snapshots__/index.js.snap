// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`conversion reducer should handle GET_REQUEST_FAIL 1`] = `
Object {
  "rates": Object {
    "data": null,
    "loader": false,
    "loadingError": undefined,
  },
}
`;

exports[`conversion reducer should handle GET_REQUEST_START 1`] = `
Object {
  "rates": Object {
    "data": null,
    "loader": true,
    "loadingError": null,
  },
}
`;

exports[`conversion reducer should handle GET_REQUEST_SUCCESS 1`] = `
Object {
  "rates": Object {
    "data": undefined,
    "loader": false,
    "loadingError": null,
  },
}
`;

exports[`conversion reducer should return the initial state 1`] = `
Object {
  "rates": Object {
    "data": null,
    "loader": false,
    "loadingError": null,
  },
}
`;
